src/components/organisms/modals/ModalRoot.tsx

```diff
@@ -1,26 +1,32 @@
-import { mapValues } from 'lodash'
 import { InstitutionSelectModalProps } from './InstitutionSelectModal/InstitutionSelectModal'
+import { DeleteRoleModalProps } from './DeleteRoleModal/DeleteRoleModal'
 import {
   lazy,
   useState,
   useCallback,
   createRef,
   useImperativeHandle,
+  Suspense,
 } from 'react'
 
 const InstitutionSelectModal = lazy(
   () => import('./InstitutionSelectModal/InstitutionSelectModal')
 )
 
+const DeleteRoleModal = lazy(() => import('./DeleteRoleModal/DeleteRoleModal'))
+
 export enum ModalTypes {
   InstitutionSelect = 'institutionSelect',
+  DeleteRole = 'deleteRole',
 }
 
 // Add other modal props types here as well
-type ModalPropTypes = Omit<InstitutionSelectModalProps, 'closeModal'>
+type ModalPropTypes = Omit<InstitutionSelectModalProps, 'closeModal'> &
+  Omit<DeleteRoleModalProps, 'closeModal'>
```

Comment:
I think "or" is better than "and" here. We don't need to include all modal props all the time